src/scripts/main.js

```diff
@@ -1,5 +1,29 @@
 'use strict';
 
 document.addEventListener('click', (e) => {
-  // write code here
+  const wall = document.querySelector('.wall');
+  const spider = wall.querySelector('.spider');
+
+  if (e.target === wall) {
+    const coords = wall.getBoundingClientRect();
+
+    spider.style.left = e.clientX - coords.left - spider.offsetWidth / 2 + 'px';
+    spider.style.top = e.clientY - coords.top - spider.offsetHeight / 2 + 'px';
+
+    if (e.clientX < coords.left + spider.offsetWidth) {
+      spider.style.left = 0 + 'px';
+    }
+
+    if (e.clientY < coords.top + spider.offsetHeight) {
+      spider.style.top = 0 + 'px';
+    }
+
+    if (e.clientX > coords.left + wall.clientWidth - spider.offsetWidth) {
+      spider.style.left = wall.clientWidth - spider.offsetWidth + 'px';
+    }
```

Comment:
This condition checks if the spider would be positioned outside the right boundary of the wall. The logic is correct, but ensure that 'coords.left', 'wall.clientWidth', and 'spider.offsetWidth' are valid numbers.